pkg/common/catalog/catalog.go

```diff
@@ -33,8 +33,9 @@ type Catalog interface {
 	// the generic Plugin type
 	Plugins() []*ManagedPlugin
 
-	// Finds plugin metadata
-	Find(Plugin) *ManagedPlugin
+	// ConfigFor finds the plugin configuration for the supplied plugin. nil
+	// is returned if the plugin is not managed by the catalog.
+	ConfigFor(interface{}) *PluginConfig
```

Comment:
Perhaps this would be more idiomatic as `ConfigFor(interface{}) (*PluginConfig, bool)`?

Reply:
works for me